src/wrappers/themis/react-native-themis/android/src/main/java/com/reactnativethemis/ThemisModule.java

```diff
@@ -163,421 +159,346 @@ public void symmetricKey(Callback callback)
   {
-    byte[] bkey;
     try {
-      bkey = dataDeserialize(symmetricKey);
-    } catch (ByteOverflowException e) {
-      errorCallback.invoke(e.getMessage());
-      return;
+      byte[] bkey = dataDeserialize(symmetricKey);
+      SecureCell.TokenProtect cell = SecureCell.TokenProtectWithKey(bkey);
+      byte[] plaintextBinary = plaintext.getBytes(StandardCharsets.UTF_8);
+      byte[] contextBinary = context.getBytes(StandardCharsets.UTF_8);
+      SecureCellData result = cell.encrypt(plaintextBinary, contextBinary);
+      byte[] encrypted = result.getProtectedData();
+      byte[] authToken = result.getAdditionalData();
+      WritableMap response = new WritableNativeMap();
+      response.putArray("encrypted", dataSerialize(encrypted));
+      response.putArray("token", dataSerialize(authToken));
+      successCallback.invoke(response);
+    } catch (Exception e) {
+      WritableMap error = new WritableNativeMap();
+      error.putString("message", e.toString());
+      errorCallback.invoke(error);
     }
-    SecureCell.TokenProtect cell = SecureCell.TokenProtectWithKey(bkey);
-    byte[] txt = plaintext.getBytes(StandardCharsets.UTF_8);
-    byte[] ctx = context.getBytes(StandardCharsets.UTF_8);
-    SecureCellData result = cell.encrypt(txt, ctx);
-    byte[] encrypted = result.getProtectedData();
-    byte[] authToken = result.getAdditionalData();
-
-    WritableMap response = new WritableNativeMap();
-    response.putArray("encrypted", dataSerialize(encrypted));
-    response.putArray("token", dataSerialize(authToken));
-
-    successCallback.invoke(response);
   }
 
   @ReactMethod
   public void secureCellTokenProtectDecrypt(ReadableArray symmetricKey,
-                                  ReadableArray encrypted,
-                                  ReadableArray token,
-                                  String context,
-                                  Callback successCallback,
-                                  Callback errorCallback)
+                                            ReadableArray encrypted,
+                                            ReadableArray token,
+                                                   String context,
+                                                 Callback successCallback,
+                                                 Callback errorCallback)
   {
-    byte[] bkey;
-    byte[] enc;
-    byte[] tkn;
-    try {
-      bkey = dataDeserialize(symmetricKey);
-      enc  = dataDeserialize(encrypted);
-      tkn  = dataDeserialize(token);
-    } catch (ByteOverflowException e) {
-      errorCallback.invoke(e.getMessage());
-      return;
-    }
-    SecureCell.TokenProtect cell = SecureCell.TokenProtectWithKey(bkey);
-    byte[] ctx = context.getBytes(StandardCharsets.UTF_8);
-
     try {
-      byte[] decrypted = cell.decrypt(enc, tkn, ctx);
+      byte[] bkey = dataDeserialize(symmetricKey);
+      byte[] enc = dataDeserialize(encrypted);
+      byte[] tkn = dataDeserialize(token);
```

Comment:
Short and telling names. I renamed txt and ctx because @Lagovas asked me. And he is right. Easy to make mistype. But bkey, enc, and tkn is not the same. But you probably want to ask me to rename it too?

Reply:
actually, I prefer when all code looks similar. Or we use fully understandable variable names, or we use short variants always. In this method we can see `symmetricKey`, `encrypted`, `contextBinary`, `decrypted` which looks like first variant. And we see `bkey` (binaryKey? base64Key? bufferedKey?), `enc` and `tkn` which look different. So much better to be consistent and use same naming approach and use readable and understandable names everywhere.

Reply:
> But you probably want to ask me to rename it too?

Yes please, let's follow a uniform style guide, writing understandable names.

Reply:
thanx!